### ISSUE TITLE
`Style/WordArray` peculiar autocorrect error

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop#11284]
Source:
```ruby
  FOO = %w[
    admin
    one@example.com
    two@example.com
    three@example.com
    four.four@gmail.com
    five@example.com
    six@example.com
    seven@example.com
    eight@example.com
    nine@example.com
    ten@example.com
  ].freeze
```

After corection:
```ruby
  FOO = [ample'admin',    s'one@example.com',    s'two@example.com',    s'three@example.com',    s'four.four@gmail.com',    s'five@example.com',    s'six@example.com',    s'seven@example.com',    s'eight@example.com',    s'nine@example.com',    s'ten@example.com'].freeze
```

--------

## Expected behavior

The resulting array is the same, just in a different notation.

## Actual behavior

Array is broken.

## Steps to reproduce the problem

See above

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.37.1 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 2.7.5) [x86_64-linux]
  - rubocop-performance 1.15.0
  - rubocop-rails 2.17.2
  - rubocop-rspec 2.14.2
```
